### ISSUE TITLE
Remove HTTP::RequestProcessor

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#7248]
The `HTTP::RequestProcessor` class simply delegates the request handling for `HTTP::Server`. I don't think this needs to be a distinct type. There is a 1:1 coupling between each server and processor, and it basically consists of one method and two ivars which can easily be integrated into `HTTP::Server` directly.

I suppose the class was originally introduced in ecdcea9b945559fdd44b4fb6f993ce20e287b207 as an easy to use interface for testing request processing, since `HTTP::Server` required to set up a TCP server for that and `RequestProcessor` operates on IOs which can be easily be tested using `IO::Memory`.

After #5776 this is no longer the case. Setting up a `HTTP::Server` instance doesn't anymore instantly open a TCP server. It could easily be used for testing handler chains without going through any network socket.

`HTTP::RequestProcessor` is part of the public API, but has no documentation. I doubt it has any practical use case where it couldn't be substituted by `HTTP::Server` directly.

**Comment by RX14** [comment]
I don't mind the split, purely on a code separation point of view. `HTTP::Server` dealing with the incmping connections and `HTTP::RequestProcessor` dealing with an individual connection's lifecycle makes sense to me.

I don't think it needs to be a documented type though.

**Comment by RX14** [comment]
@straight-shoota yes, it's a single method, but it's a long one and I wouldn't want to make `http/server.cr` longer. Any complaints of mine are purely stylistic.

**Comment by straight-shoota** [comment]
Oh, yes. I'd actually too prefer to leave the code in `request_processor.cr` and just reopen `HTTP::Server` instead of creating a new class.
